### PR TITLE
Added v4 shooting rule and fixed v3 shooting rule

### PR DESCRIPTION
**Pull request by TrevorBrunette**

### CODE REVIEW

File: src/main/java/pro/trevor/tankgame/rule/impl/version3/Ruleset.java [hunk 1]
```diff
@@ -85,7 +85,10 @@ public void registerPlayerRules(RulesetDescription ruleset) {
         playerRules.put(Tank.class, PlayerRules.GetShareGoldWithTaxRule(1));
         playerRules.put(Tank.class, PlayerRules.SPEND_ACTION_TO_MOVE);
         playerRules.put(Tank.class, PlayerRules.SpendActionToShootWithDeathHandle(LineOfSight::hasLineOfSightV3,
-            (s, t, d) -> t.setGold(t.getGold() + d.getGold() + d.getBounty())
+            (s, t, d) -> {
+                t.setGold(t.getGold() + d.getGold() + d.getBounty());
+                s.getCouncil().setCoffer(s.getCouncil().getCoffer() + 1);
```
Comment (BryanFriestad) [inline src/main/java/pro/trevor/tankgame/rule/impl/version3/Ruleset.java:90]: I don't think the council gets anything upon looting in v3. Not sure why this is being changed.

File: src/main/java/pro/trevor/tankgame/rule/impl/version4/Ruleset.java [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
+
+        playerRules.put(Tank.class, PlayerRules.SpendActionToShootWithDeathHandle(LineOfSight::hasLineOfSightV4,
+            (s, t, d) -> {
+                t.setGold(t.getGold() + d.getBounty());
+                switch (d.getGold()) {
+                    case 0 -> {}
+                    case 1 -> t.setGold(t.getGold() + 1);
+                    default -> {
+                        // Tax is target tank gold * 0.25 rounded up
+                        int tax = (d.getGold() + 3) / 4;
```
Comment (BryanFriestad) [inline src/main/java/pro/trevor/tankgame/rule/impl/version4/Ruleset.java:25]: This logic is pretty complex. I'm going to review it some more to make sure it's right, but you need to add some unit tests for this as well. Focus on the looting aspect, but really all parts of shooting should be handled. This is also why I moved all rule logic into functions or static final variables, so they could be fully unit tested. Right now, with this logic being in the ruleset, it cannot be tested.

**Review by BryanFriestad — CHANGES_REQUESTED** [review]

File: src/main/java/pro/trevor/tankgame/rule/impl/version3/Ruleset.java (hunk 1 above)
Comment (TrevorBrunette) [reply src/main/java/pro/trevor/tankgame/rule/impl/version3/Ruleset.java]: DONE

File: src/main/java/pro/trevor/tankgame/rule/impl/version4/Ruleset.java (hunk 2 above)
Comment (TrevorBrunette) [reply src/main/java/pro/trevor/tankgame/rule/impl/version4/Ruleset.java]: Tested and DONE

**Review by BryanFriestad — APPROVED** [review]

**Pull request merged**